### ISSUE TITLE
[1.15.2] CRASH - OpenJ9 JVM

### ISSUE THREAD

**Issue by jrkulmer** [Cryptic-Mushroom/The-Midnight#404]
The Midnight - 0.5.7-3
Forge - 31.2.0
OpenJ9 - 14.0.1.7 (no command line args other than -Xmx32G)

x64 PC.

Looks like a bug not caught by the bundled JVM, as the crash also occurs when using JDK 14 with ZGC garbage collector. Same exception.

Crash on startup: https://gist.github.com/Jonathing/d63e01a2ceb9965f69d3e9c188f72a77

**Comment by Jonathing** [comment]
Edited message by putting the log in a gist. The log file was too big for me to read properly within GitHub issues.

**Comment by Jonathing** [comment]
I am closing this issue for the following reasons.
1. We have not yet updated The Midnight for the latest recommended version of Minecraft Forge. Although the current version could work, it is possible that it may not. However, this leads into my next point.
2. You are running Minecraft on Java 9 and Java 14, both versions which are higher than Java 8. The Midnight mod is written specifically for Java 8.

If for some reason you don't have Java 8 installed on your computer, I personally recommend using [Amazon Corretto 8](https://docs.aws.amazon.com/corretto/latest/corretto-8-ug/downloads-list.html) as it is fully functional with Minecraft Forge on Minecraft 1.13+ and contains various bug fixes that normal Java does not have. When we begin working on The Midnight Rewrite for Minecraft 1.16, we may take Java 11 compatibility into consideration, but it is highly likely that we will continue to stick with Java 8 as that is what Minecraft itself is built on.

tl;dr use Java 8 please

**Comment by jrkulmer** [comment]
I'm aware its built for Java 8 - I'm not complaining that I can't play it with a different version of Java, don't worry. I just wondered whether the other VMs had picked up an error uncaught by the Java 8 runtime, which may be useful long term if it is an actual bug.

**Comment by Jonathing** [comment]
Yeah, it's specifically a Java 8 thing. The error happens with a lambda and that's something introduced in Java 8. I don't know if they changed how it worked in Java 9.
